app/customer.py

```diff
@@ -0,0 +1,77 @@
+from dataclasses import dataclass
+
+from app.car import Car
+from app.shop import Shops
+
+
+@dataclass
+class Customers:
+    name: str
+    product_cart: dict
+    location: list
+    money: int
+    car: Car
+
+    def __post_init__(self) -> None:
+        if isinstance(self.car, dict):
+            self.car = Car(**self.car)
+
+    def customers_info(self) -> None:
+        print(f"{self.name} has {self.money} dollars")
+
+    def calculate_distance(self, shop: Shops) -> float:
+        return ((self.location[0] - shop.location[0]) ** 2
+                + (self.location[1] - shop.location[1]) ** 2) ** 0.5
+
+    def calculate_trip_cost(self, fuel_price: float, shop: Shops) -> float:
+        distance = self.calculate_distance(shop)
+        fuel_consumption = self.car.fuel_consumption_per_100_km(distance)
+        return (fuel_consumption * fuel_price) * 2
+
+    def look_for_cheapest_shop(self,
+                               fuel_price: float,
+                               shops: list[Shops]
+                               ) -> tuple[None | Shops, float, float]:
+        cheapest_shop = None
+        min_product_cost = None
+        min_total_cost = float("inf")
+
+        for shop in shops:
+            trip_cost = self.calculate_trip_cost(fuel_price, shop)
+            products_cost = shop.calculate_products_cost(self.product_cart)
+            total_cost = trip_cost + products_cost
+            print(f"{self.name}'s trip to the {shop.name} "
+                  f"costs {round(total_cost, 2)}")
+
+            if total_cost <= self.money and total_cost < min_total_cost:
+                cheapest_shop = shop
+                min_product_cost = products_cost
+                min_total_cost = total_cost
+
+        return cheapest_shop, min_total_cost, min_product_cost
+
+    def make_purchase(self,
+                      shop: Shops,
+                      total_cost: float,
+                      product_cost: float) -> None:
+        print(f"{self.name} rides to {shop.name}")

```

Comment:
when customer rides to shop his location should be changed to shop location and the same with home